hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -755,21 +763,30 @@ onOpenNetworkReqDec env ledger ttl openState decommitTx =
   waitOnApplicableDecommit cont =
     case mExistingDecommitTx of
       Nothing ->
-        case applyTransactions currentSlot localUTxO [decommitTx] of
-          Right utxo' -> cont utxo'
-          Left (_, validationError)
-            | ttl > 0 ->
-                wait $
-                  WaitOnNotApplicableDecommitTx
-                    ServerOutput.DecommitTxInvalid{localUTxO, validationError}
-            | otherwise ->
-                cause . ClientEffect $
-                  ServerOutput.DecommitInvalid
-                    { headId
-                    , decommitTx
-                    , decommitInvalidReason =
-                        ServerOutput.DecommitTxInvalid{localUTxO, validationError}
-                    }
+        if utxoFromTx decommitTx == mempty
```

Comment:
Instead of checking it here, I would suggest doing it as "a step before" like we do in other places. See also the TODO in line 735/744 which mentions the #1502 issue.

Do we need to update the spec?